compiler/src/openxla/compiler/nvgpu/Conversion/ConvertHLOToCUDNN.cpp

```diff
@@ -58,107 +59,51 @@ static FailureOr<Layout> getCudnnTensorLayout(int64_t batchDim,
 }
 
 static FailureOr<Layout> getCudnnKernelLayout(int64_t inputDim,
-                                              int64_t outputDim,
-                                              MLIRContext* ctx) {
+                                              int64_t outputDim) {
   if (outputDim != 0) return failure();
   // Return input/output layout instead of the kernel layout so that casts
   // cancel themselves. The actual layouts are the same and the lowering and
-  // runtime implementation handle it correctly. TODO(chsigg): revert when
-  // layout propagation is implemented.
+  // runtime implementation handle it correctly.
+  // TODO(chsigg): revert when layout propagation is implemented.
   if (inputDim == 1) return Layout::NCHW;  // KCHW
   if (inputDim == 3) return Layout::NHWC;  // KHWC
   return failure();
 }
 
-// Outlines all transitive ops defining 'result' into a cudnn.graph op and calls
-// it with `arguments`.
-static LogicalResult outlineToGraph(TensorValue result,
-                                    ArrayRef<TensorValue> arguments,
-                                    FunctionType funcType,
-                                    PatternRewriter& rewriter) {
-  Operation* root = result.getDefiningOp();
-  if (!root)
-    return rewriter.notifyMatchFailure(result.getLoc(), "expected def by op");
-  if (root->getNumResults() != 1)
-    return rewriter.notifyMatchFailure(result.getLoc(), "expected one result");
-  func::FuncOp func = root->getParentOfType<func::FuncOp>();
-  if (!func) return rewriter.notifyMatchFailure(root, "expected child of func");
-  ModuleOp m = func->getParentOfType<ModuleOp>();
-  if (!m)
-    return rewriter.notifyMatchFailure(root,
-                                       "expected child of func in module");
-
-  // Collect the set of ops to clone into the region.
-  SetVector<Operation*> ops;
-  ops.insert(root);
-  DenseSet<Value> argSet(arguments.begin(), arguments.end());
-  for (size_t index = 0; index < ops.size(); ++index) {
-    Operation* op = ops[index];
-    for (Value operand : op->getOperands()) {
-      if (argSet.count(operand)) continue;
-      Operation* def = operand.getDefiningOp();
-      if (!def)
-        return rewriter.notifyMatchFailure(op, "expected operands def by op");
-      ops.insert(def);
-    }
+class CUDNNHandleCache {
+ public:
+  IREE::Util::GlobalOp getGlobalHandle(PatternRewriter& rewriter, Location loc,
+                                       ModuleOp m, StringRef baseName) {
```

Comment:
given that we'll have one handle per-module, I think it's ok to skip base name, as it will be confusin if `convolution.handle` will be used for non-convolution graphs

Reply:
Done.